pysimgrid/simdag/algorithms/ldcp.py

```diff
@@ -0,0 +1,393 @@
+    )
+    self._schedulable_tasks = set(
+      task for task in self._nxgraph.nodes_iter()
+      if self._nxgraph.node[task]['num_unscheduled_parents'] == 0
+    )
+    self._scheduled_tasks = set()
+    self._unscheduled_tasks = set(self._nxgraph.nodes_iter())
+    self._temporary_edges = {host: set() for host in self.hosts}
+    self._dagp = {host: self._simulation.get_task_graph() for host in self.hosts}
+    for host, graph in self._dagp.items():
+      networkx.set_node_attributes(
+        graph,
+        'size',
+        {task: task.amount / host.speed for task in graph}
+      )
+    self._urank = {host: {} for host in self.hosts}
+    self.update_urank(initial=True)
+    self._task_to_schedule = None
+    self._host_to_schedule = None
+    self._last_identified_task = None
+    self._last_used_host_dagp = None
+
+  @property
+  def hosts(self):
+    """
+    :return: list of hosts in simulation system
+    """
+    container_type = type(self._simulation.hosts)
+    return container_type(host for host in self._simulation.hosts if not cscheduling.is_master_host(host))
+
+  @property
+  def mean_bandwidth(self):
+    """
+    :return: mean bandwidth in simulation system
+    """
+    return self._platform_model.bandwidth.mean()
+
+  @property
+  def timetable(self):
+    """
+    :return: timetable for available hosts
+    """
+    return self._state.timetable
+
+  @staticmethod
+  def pair_to_key(pair):
+    """
+    :param pair: (key, value)
+    :return: key
+    """
+    return operator.itemgetter(0)(pair)
+
+  @staticmethod
+  def pair_to_value(pair):
+    """
+    :param pair: (key, value)
+    :return: value
+    """
+    return operator.itemgetter(1)(pair)
+
+  @staticmethod
+  def timesheet_to_tasks(timesheet):
+    """
+    :param timesheet: timesheet of scheduled tasks for some host: [(task, start, finish), ...]
+    :return: generator of tasks obtained by removing start and finish times: [task1, task2, ...]
+    """
+    return map(operator.itemgetter(0), timesheet)
+
+  def get_schedule(self, simulation):
+    """
+    Overridden method that computes timetable for scheduling and expected makespan.
+    :param simulation: description of simulating system
+    :return: tuple that contains scheduling plan and expected makespan (sec.)
+    """
+    for _ in self._nxgraph.nodes_iter():
+      self.select_task_to_schedule()
+      if self.try_schedule_boundary_task():
+        continue
+      pos, start, finish = self.select_host_to_schedule()
+      self.assign_task_to_host(pos, start, finish)
+      self.update_size_wrt_selected_task()
+      self.update_communications_costs()
+      self.update_execution_constraints()
+      self.update_zero_cost_edges_on_dagp_wrt_selected_host()
+      self.update_urank()
+    expected_makespan = max(state['ect'] for state in self._state.task_states.values())
+    return self._state.schedule, expected_makespan
+
+  def get_uras_with_term(self, task, host, use_only_unscheduled=False):
```

Comment:
Default value looks strange. Method called only once and this call uses a non-default value.